### PR TITLE
improve default behavior of the module, fixes #2

### LINKED ISSUE #2: improve default behavior of the module
I think we could do the following improvements:

- [ ] Capture metrics regardless of having a current trace context or not.
- [ ] We have some regular expressions there to detect the type of query, but at the end we are just storing metrics for reads and writes, we could get rid of those regex evaluations and measure calls to `java.sql.PreparedStatement#executeQuery` and `java.sql.PreparedStatement#executeUpdate` to get the same metrics.
- [ ] Include a minMaxCounter that tracks how many operations are currently executing against the database.
- [ ] Add the most common JDBC packages drivers into the default `aop.xml` to ensure that for most people it will just work out of the box.
- [ ] Add a configuration to decide whether the user wants trace-segments to be generated for the tracked JDBC calls, if so we should avoid using special characters in the default names (e.g. `Jdbc[Select]`).

Another thing that might be cool is to be able to separate metrics by connection pool, but that can come later.

### PR DESCRIPTION
**Pull request by ivantopo**
This PR aims to solve the issues described in #2. I went full retarded and added support for connection pools right away, with support for Hikari.. I'll add support for TomcatDBCP and BoneCP to this PR to ensure that the approach is fine and works with all of them before getting this merged.

### CODE REVIEW

**Comment by ivantopo** [comment]
snapshot published here: https://bintray.com/kamon-io/snapshots/kamon-jdbc/0.6.6-156d4ab57980130828dbb217d5f562a8e29ed28c

File: src/main/scala/kamon/jdbc/instrumentation/StatementInstrumentation.scala [hunk 1]
```diff
@@ -15,104 +15,146 @@
...
         JdbcExtension.processSlowQuery(sql, timeSpentInMillis)
       }
     }
   }
 
-  def recordWrite(pjp: ProceedingJoinPoint, sql: String)(implicit statementRecorder: StatementsMetrics): Any = {
-    withTimeSpent(pjp.proceedWithErrorHandler(sql)) { timeSpent ⇒
-      statementRecorder.writes.record(timeSpent)
-    }
+  def withSegment[A](ctx: TraceContext, statementType: String, statement: String)(thunk: ⇒ A): A = {
```
Comment (dpsoft) [inline src/main/scala/kamon/jdbc/instrumentation/StatementInstrumentation.scala:145]: I think we need drive the case when the segment is finish with error and finish the segment with:`segment.finishWithError(Throwable)`. @ivantopo WDYT?

File: src/main/scala/kamon/jdbc/instrumentation/HikariInstrumentation.scala [hunk 2]
```diff
@@ -0,0 +1,114 @@
+package kamon.jdbc.instrumentation
```
Comment (dpsoft) [inline src/main/scala/kamon/jdbc/instrumentation/HikariInstrumentation.scala:1]: Copyright headers ;)

**Review by dpsoft — COMMENTED** [review]
@ivantopo I really like this!!!

File: src/main/resources/META-INF/aop.xml [hunk 3]
```diff
@@ -3,10 +3,17 @@
...
-   </weaver>
+
+  <weaver>
+    <include within="com.zaxxer.hikari.pool.HikariPool"/>
+    <include within="com.zaxxer.hikari.pool.ProxyConnection"/>
+
+    <!-- JDBC Drivers -->
+    <include within="org.h2..*"/>
+    <include within="org.postgresql.jdbc4..*"/>
+    <include within="com.microsoft.sqlserver.jdbc..*"/>
```
Comment (dpsoft) [inline src/main/resources/META-INF/aop.xml]: I think is a good idea also include the `MariaDB(org.mariadb.jdbc..)` and `MySQL(com.mysql.jdbc..)` drivers. @ivantopo WDYT?

Comment (ivantopo) [reply src/main/resources/META-INF/aop.xml]: of course! I just didn't have the packages at hand.. also, I'm not sure if there is a way in which we could automate testing those :/

**Pull request merged**